### PR TITLE
Removes an unsed proc from smes.dm and cleans up the file

### PR DESCRIPTION
**Pull request by Burzah**
## What Does This PR Do
This removes an unused proc `ion_act` and cleans up the rest of the file.
Comments within the file have also been adjusted where appropriate, including the use of autodoc. 
Used defines in places where appropriate. 

(I was working on something else, but then this happened)

## Why It's Good For The Game
What is good for the code base is good for the game.

## Testing
Started an instance and confirmed that the SMES was still working as intended.

## Changelog
NPFC

### CODE REVIEW

File: code/modules/power/smes.dm [hunk 1]
```diff
@@ -420,32 +425,6 @@
...
-			for(var/mob/M in viewers(src))
-				M.show_message("<span class='warning'>[src] is making strange noises!</span>", 3, "<span class='warning'>You hear sizzling electronics.</span>", 2)
-			sleep(10*pick(4,5,6,7,10,14))
-			var/datum/effect_system/smoke_spread/smoke = new
-			smoke.set_up(3, FALSE, loc)
-			smoke.attach(src)
-			smoke.start()
-			explosion(src.loc, -1, 0, 1, 3, 1, 0)
-			qdel(src)
-			return
```
Comment (Sirryan2002) [inline code/modules/power/smes.dm:435]: quality content

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/power/smes.dm (hunk 1 above)
Comment (Burzah) [reply code/modules/power/smes.dm:435]: Truly a function of all time.

File: code/modules/power/smes.dm [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
+	/// TRUE = actually outputting, FALSE = not outputting
+	var/outputting = TRUE
+	/// Amount of power the SMES attempts to output
+	var/output_level = 50000
+	/// Cap on output_level
+	var/output_level_max = 200000
+	/// Amount of power actually outputted. may be less than output_level if the powernet returns excess power
+	var/output_used = 0
 	var/name_tag = null
 	var/obj/machinery/power/terminal/terminal = null
```
Comment (DGamerL) [inline code/modules/power/smes.dm:36]: Can you autodoc these anyways? And remove the `= null`

File: code/modules/power/smes.dm [hunk 3]
```diff
@@ -148,68 +152,70 @@
...
+		var/temporary_direction = get_dir(user, src)
+		switch(temporary_direction)
 			if(NORTHEAST, SOUTHEAST)
-				tempDir = EAST
+				temporary_direction = EAST
 			if(NORTHWEST, SOUTHWEST)
-				tempDir = WEST
-		var/turf/tempLoc = get_step(src, REVERSE_DIR(tempDir))
-		if(isspaceturf(tempLoc))
+				temporary_direction = WEST
```
Comment (DGamerL) [inline code/modules/power/smes.dm:161]: ... why do we disable diagonal clicking in line 130 but have a failsave here for diagonals?

File: code/modules/power/smes.dm (hunk 2 above)
Comment (Burzah) [reply code/modules/power/smes.dm:36]: Oh I missed those. Yeah sure I will squash those real quick.

File: code/modules/power/smes.dm (hunk 3 above)
Comment (Burzah) [reply code/modules/power/smes.dm:161]: Discussed on Discord!

**Review by DGamerL — APPROVED** [review]

**Pull request merged**